### PR TITLE
Logic/calendar matrix

### PR DESCRIPTION
**Pull request by i786m**
This pull request introduces two new utility functions for calendar calculations and adds comprehensive tests for them. The previous placeholder content in `common.mjs` and its test has been replaced with real functionality for computing the nth weekday of a month and generating a month calendar matrix.

Calendar utility functions:

* Added `getNthDayOfMonth` to compute the Date object for the nth occurrence of a specified weekday in a given month and year, including validation and error handling for invalid input.
* Added `getCalendarMatrix` to generate a matrix representing the calendar for a month, with each week as an array of day numbers or null for padding.

Testing:

* Replaced the old greeting test with tests for both `getNthDayOfMonth` and `getCalendarMatrix`, covering normal cases, edge cases, and leap years in `common.test.mjs`.

### CODE REVIEW

File: common.mjs [hunk 1]
```diff
@@ -1,5 +1,169 @@
...
+		: occurrence === 'second' ? 2
+		: occurrence === 'third' ? 3
+		: occurrence === 'fourth' ? 4
+		: occurrence === 'fifth' ? 5
+		: null;
+	if (nth === null) {
+		throw new Error('Invalid occurrence: ' + occurrence);
+	}
+	// calculate zero-based index for month and day
+	const months = [
```
Comment (Zobeir-Rigi) [inline common.mjs]: You can move months to a constant outside the functions so other functions can reuse it. This avoids repeating the same code. Like :
const MONTHS = [...];
